japicmp/src/main/java/japicmp/output/markdown/MarkdownOutputGenerator.java

```diff
@@ -0,0 +1,606 @@
+package japicmp.output.markdown;
+
+import static japicmp.model.JApiChangeStatus.*;
+import static japicmp.output.markdown.Markdown.*;
+import static japicmp.util.JApiClassFileFormatVersionHelper.*;
+import static japicmp.util.MemberValueHelper.formatMemberValue;
+import static japicmp.util.ModifierHelper.*;
+import static japicmp.util.OptionalHelper.N_A;
+import static japicmp.util.TypeNameHelper.*;
+import static java.lang.String.format;
+import static java.util.Collections.*;
+import static java.util.stream.Collectors.*;
+
+import japicmp.cmp.JApiCmpArchive;
+import japicmp.config.*;
+import japicmp.filter.Filter;
+import japicmp.model.*;
+import japicmp.model.JApiJavaObjectSerializationCompatibility.JApiJavaObjectSerializationChangeStatus;
+import japicmp.output.*;
+import japicmp.output.markdown.config.MarkdownOptions;
+import japicmp.output.semver.SemverOut;
+import japicmp.util.Optional;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import javassist.bytecode.annotation.MemberValue;
+
+public class MarkdownOutputGenerator extends OutputGenerator<String> {
+
+	final MarkdownOptions md;
+	final MarkdownReferences references = new MarkdownReferences();
+
+	public MarkdownOutputGenerator(MarkdownOptions mdOptions, List<JApiClass> jApiClasses) {
+		super(mdOptions.options, jApiClasses);
+		md = mdOptions;
+	}
+
+	public MarkdownOutputGenerator(Options options, List<JApiClass> jApiClasses) {
+		this(MarkdownOptions.newDefault(options), jApiClasses);
+	}
+
+	@Override
+	public String generate() {
+		return toString();
```

Comment:
Why override `toString()` here?

Reply:
Got it. I simplified it here: https://github.com/siom79/japicmp/pull/405/commits/e00fd6afa0cb73dd8a4d3034c058c2196eb96cc6